ui/src/submissions/literature/components/LiteratureSubmission.jsx

```diff
@@ -37,60 +43,75 @@ const FORMS_BY_DOC_TYPE = {
   },
 };
 
-class LiteratureSubmission extends Component {
-  constructor(props) {
-    super(props);
+function LiteratureSubmission({
+  error = null,
+  docType,
+  initialFormData = null,
+  onSubmit,
+}) {
+  const { component, schema, defaultData } = FORMS_BY_DOC_TYPE[docType];
+  const initialValues = useMemo(
+    () => ({ ...defaultData, ...initialFormData }),
+    [defaultData, initialFormData]
+  );
 
-    this.onFormikSubmit = this.onFormikSubmit.bind(this);
-  }
+  const [isInitilValid, setInitialValid] = useState(false);
+  useEffect(
+    () => {
+      const validate = async () => {
+        setInitialValid(await schema.isValid(initialValues));
+      };
+      validate();
+    },
```

Comment:
mann it's too complicated to read :(

Reply:
yeah, I agree, `async` calls in effect hook is pretty weird. I'll create a custom hook for this, to simplify here.